### PR TITLE
Added section on BoostCon & C++Now to History topic

### PR DESCRIPTION
**Pull request by PeterTurcan**

### CODE REVIEW

File: user-guide/modules/ROOT/pages/boost-history.adoc [hunk 1]
```diff
@@ -25,6 +25,34 @@ One of the defining characteristics of Boost is its strict peer review process.
 
 Boost has had a profound impact on the pass:[C++] language and its community. Many of the libraries and concepts introduced by Boost have been adopted into the pass:[C++] Standard Library, including smart pointers, regular expressions, and the boost:lambda[] function syntax. Boost has also been a fertile ground for experimenting with new ideas and techniques, which have later made their way into the pass:[C++] language itself, such as boost:type_traits[] and boost:mpl[].
 
+=== Developer Conference
+
+_BoostCon_, initiated in 2007, was a conference dedicated to the discussion and improvement of both Boost libraries and pass:[C++] software development in general. In 2012, BoostCon was rebranded as _pass:[C++] Now_ to reflect a broader scope. While Boost libraries remained a crucial part of the discussion, the conference expanded to cover popular and advanced pass:[C++] techniques.
```
Comment (alandefreitas) [inline user-guide/modules/ROOT/pages/boost-history.adoc:30]: Are there links we can include here?

Comment (PeterTurcan) [reply user-guide/modules/ROOT/pages/boost-history.adoc]: @alandefreitas  added links to Boostcon 2012, cppnow

File: user-guide/modules/ROOT/pages/boost-history.adoc [hunk 2]
```diff
@@ -25,28 +25,56 @@ One of the defining characteristics of Boost is its strict peer review process.
...
+* https://lists.boost.org/Archives/boost//2011/02/177137.php[Boost Filesystem now has public GitHub repository]
+
+
 == A Timeline of Boost Exclusivity
 
 While many Boost libraries have been incorporated into the C++ Standard Library, most others remain exclusive to Boost, and are highly regarded for their capabilities and performance. Some of these noteworthy libraries include:
 
 [square]
-* boost:graph[] : An extensive library for graph data structures and algorithms, enabling developers to work with graphs and network structures, while providing efficient implementations of popular graph algorithms like Dijkstra's, Kruskal's, and more. Released in *August 2000*.
+* boost:smart_ptr[] : Smart pointers automatically delete the object pointed to at the appropriate time. Development of this library started in 1994 with updates made as recent as 2019. Developers include Greg Colvin, Beman Dawes, Peter Dimov, Darin Adler and Glen Fernandes. This library was one of the original 24 in the first release of Boost in *September 1999*.
```
Comment (alandefreitas) [inline user-guide/modules/ROOT/pages/boost-history.adoc:59]: This one has been incorporated into the C++ Standard Library.

File: user-guide/modules/ROOT/pages/boost-history.adoc [hunk 3]
```diff
@@ -25,28 +25,56 @@ One of the defining characteristics of Boost is its strict peer review process.
...
 
-* boost:compute[]: A library for parallel and high-performance computing on GPUs and other OpenCL devices, offering a high-level, easy-to-use interface for working with OpenCL. Added to Boost *November 2014*.
+* boost:geometry[] : A library providing a collection of algorithms and data structures for working with geometrical objects, such as points, lines, polygons, and more. It also supports spatial indexing, coordinate system transformations, and various distance calculations. Developed by Barend Gehrels, Bruno Lalande, Mateusz Loskot, Adam Wulkiewicz, Menelaos Karavelas and Vissarion Fisikopoulos and added to Boost *July 2011*.
 
-* boost:hana[] : A modern metaprogramming library for pass:[C++] that employs cutting-edge pass:[C++] features to facilitate powerful and expressive metaprogramming techniques, such as heterogeneous containers, compile-time algorithms, and type introspection. Added to Boost *May 2016*.
+* boost:algorithm[]: A library offering a collection of general purpose algorithm. Developed by Marshall Clow and added to Boost *June 2012*.
```
Comment (alandefreitas) [inline user-guide/modules/ROOT/pages/boost-history.adoc:69]: So has this one.

File: user-guide/modules/ROOT/pages/boost-history.adoc [hunk 4]
```diff
@@ -67,6 +99,12 @@ _The best software solutions are written using the C++ programming language_
 [circle]
 - https://www.boost.org/users/proposal.pdf[Proposal for a C++ Library Repository Web Site]
 
+
+=== Boost Releases
+
+[circle]
+- https://www.boost.org/users/history/[Boost Version History]
```
Comment (alandefreitas) [inline user-guide/modules/ROOT/pages/boost-history.adoc:106]: Is this link valid on the new website?

Comment (PeterTurcan) [reply user-guide/modules/ROOT/pages/boost-history.adoc:106]: Not sure, will remove it for now.

File: user-guide/modules/ROOT/pages/boost-history.adoc (hunk 3 above)
Comment (PeterTurcan) [reply user-guide/modules/ROOT/pages/boost-history.adoc:69]: OK, made a note of this for both libraries.

**Comment by PeterTurcan** [comment]
@alandefreitas Made your suggested changes and squashed

File: user-guide/modules/ROOT/pages/boost-history.adoc (hunk 3 above)
Comment (alandefreitas) [reply user-guide/modules/ROOT/pages/boost-history.adoc]: Boost heap (priority queues) were incorporated in the standard library https://en.cppreference.com/w/cpp/container/priority_queue

**Comment by alandefreitas** [comment]
See https://github.com/cppalliance/site-docs/pull/146#discussion_r1380897265

**Comment by PeterTurcan** [comment]
@alandefreitas Removed reference to Heap lib, and squashed

**Pull request merged**